flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/AbstractHeapState.java

```diff
@@ -112,6 +115,29 @@ public byte[] getSerializedValue(
         return KvStateSerializer.serializeValue(result, safeValueSerializer);
     }
 
+    public SV migrateValueForTtlStateMigration(
+            SV state,
+            TypeSerializer<SV> priorSerializer,
+            TypeSerializer<SV> newSerializer,
+            TtlTimeProvider ttlTimeProvider) {
+
+        Preconditions.checkArgument(priorSerializer instanceof TtlAwareSerializer);
```

Comment:
nit: in the variable names we use  prior and new and previous and current. I suggest being consistent; 
previousSerializer previousAwareSerializer
currentSerializer currentTtlAwareSerializer